server/server/views/registration.py

```diff
@@ -0,0 +1,66 @@
+"""Views for registration user and confirm registration user in system"""
+from datetime import datetime
+
+from cornice import Service
+from passlib.hash import pbkdf2_sha256
+from passlib.totp import generate_secret
+from pyramid.view import view_config
+from pyramid_mailer.mailer import Mailer
+from pyramid_mailer.message import Message
+from pyramid.response import Response
+from pyramid.httpexceptions import HTTPNotFound
+
+from server.models.user import User
+from server.models.user_status import UserStatus
+from server.models.role import Role
+
+
+register = Service(name='registration', path='/registration')
+confirm_register = Service(name='email_confirm', path='/email_confirm/{email_confirm}')
+
+
+@register.post()
+def registration_view(request):
+    """Registration view
+
+    This function get email and password from json request, check if email
+    isn't in db - add email and password into db, set status_id to 'Non_active',
+    generate and send url-token to user email.
+    """
+    json = request.json_body
+    user_query = User.get_one(request, email=request.json_body['email'])
+    if user_query is None:
+        User.add_user(request, email=request.json['email'],
+                      password=pbkdf2_sha256.hash(request.json['password']),
+                      url_token=generate_secret(),
```

Comment:
You can save url_token in some variable before it and you will not need to make a request at line 38.

Reply:
Dear Merteg, thank you for your comment, but in this case I don't want to follow your advice.